src/screens/OtherProfile/index.js

```diff
@@ -179,22 +173,13 @@ const OtherProfile = () => {
     };
   }, []);
 
-  React.useEffect(() => {
-    initData();
-  }, []);
-
   React.useEffect(() => {
     getOtherFeeds();
   }, [other_id]);
 
-  const initData = () => {
-    setUserId(params.data.user_id);
-    setUsername(params.data.username);
-  };
-
   const onShare = async () => {
     onShareUserButtonClicked();
-    ShareUtils.shareUserLink(username);
+    ShareUtils.shareUserLink(params?.data?.username);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The removal of the `initData()` function and its call in the `useEffect` hook is a good change if the initialization of `user_id` and `username` is now handled inside the `useOtherProfileScreenHooks()` hook. The use of `params?.data?.username` in the `onShare` function instead of the previously set local state `username` is also a good change for consistency.



<!-- This is an auto-generated comment by OSS CodeRabbit -->